metals/src/main/scala/scala/meta/internal/metals/StatusBar.scala

```diff
@@ -107,9 +115,34 @@ final class StatusBar(
       showTimer: Boolean = false,
       progress: Option[TaskProgress] = None,
   ): Future[T] = {
-    items.add(Progress(message, value, showTimer, progress))
-    tickIfHidden()
-    value
+
+    if (clientConfig.statusBarState == StatusBarState.Off) {
+      val uuid = UUID.randomUUID().toString()
+      val token = messages.Either.forLeft[String, Integer](uuid)
+
+      val begin = new WorkDoneProgressBegin()
+      begin.setTitle(message)
+      val notification =
+        messages.Either.forLeft[WorkDoneProgressNotification, Object](
+          begin
+        )
+
+      client.createProgress(new WorkDoneProgressCreateParams(token))
```

Comment:
This does return a `CompletableFuture[Unit]` which currently I'm throwing away.  Not clear what this future represents and hence if it should be blocked on.